src/journal/compress.c

```diff
@@ -541,6 +645,90 @@ int compress_stream_lz4(int fdf, int fdt, uint64_t max_bytes) {
 #endif
 }
 
+int compress_stream_zstd(int fdf, int fdt, uint64_t max_bytes) {
+#if HAVE_ZSTD
+        _cleanup_(ZSTD_freeCCtxp) ZSTD_CCtx *s = ZSTD_createCStream();
+        size_t out_size = ZSTD_CStreamOutSize();
+        uint8_t out[out_size];
+        char *src = NULL;
+        struct stat st;
+        ZSTD_inBuffer ZSTD_in = {src, 0, 0};
+        ZSTD_outBuffer ZSTD_out = {&out, sizeof(out), 0};
```

Comment:
I'd prefer to have the chain of `out_size`, `out`, and `ZSTD_out` on consecutive lines, preferably at the end of the variable declaration list.